src/PropertyAppTests/ModelHandlingTests/ModelHandlerTests.cs

```diff
@@ -0,0 +1,152 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using PropertyApp.Model;
+using PropertyApp.ModelHandling;
+using System.Diagnostics.CodeAnalysis;
+
+namespace PropertyAppTests.ModelHandlingTests;
+
+[ExcludeFromCodeCoverage]
+[TestFixture]
+public class ModelHandlerTests
+{
+    private ModelHandler _modelHandler;
+    private IServiceProvider _serviceProvider;
+
+    [SetUp]
+    public void SetUp()
+    {
+        var services = new ServiceCollection();
+        services.AddSingleton<ModelHandler>();
+        _serviceProvider = services.BuildServiceProvider();
+        _modelHandler = _serviceProvider.GetRequiredService<ModelHandler>();
+    }
+
+    [TearDown]
+    public void TearDown()
+    {
+        _modelHandler.ClearModel();
+    }
+
+    [Ignore("TODO: Need to learn implementing singleton pattern properly, come back to this one")]
+    [Test]
+    public void ThereCanBeOnlyOne()
+    {
+        // Act
+        var attemptToCreateSecondInstance = () =>
+        {
+            var kurganInstance = new ModelHandler();
+        };
+
+        // Assert
+        attemptToCreateSecondInstance.Should().Throw<InvalidOperationException>()
+            .WithMessage("Cannot create more than one instance of ModelHandler class");
+    }
+
+    [Test]
+    public void SetImagePathSetsTheImagePath()
+    {
+        // Arrange
+        _modelHandler.InitModel();
```

Comment:
If `ClearModel()` is in teardown method then `InitModel()` should be in setup method

Reply:
Can't because some tests rely on init model not being called